core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -143,7 +143,7 @@
     public static final Setting<Boolean> NODE_INGEST_SETTING =
         Setting.boolSetting("node.ingest", true, Property.NodeScope);
     public static final Setting<String> NODE_NAME_SETTING = Setting.simpleString("node.name", Property.NodeScope);
-    // this sucks that folks can mistype client etc and get away with it.
+    // this sucks that folks can mistype data, master or ingest and get away with it.
```

Comment:
I don;t think this would have made this specific PR easier, it would remove a weird if that's it. I would do that as a follow-up, don't want this PR to get crazy big.